pandas/tools/pivot.py

```diff
@@ -136,8 +136,14 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
 
     table = agged
     if table.index.nlevels > 1:
-        to_unstack = [agged.index.names[i] or i
-                      for i in range(len(index), len(keys))]
+        index_names = agged.index.names[:len(index)]
```

Comment:
ahh, ok, so these should *always* use positionals then; you might need to change the downstream ``_get_level_values()`` (always positional) rather than ``get_level_values()`` (figures it out)